libs/androidlib/src/mill/androidlib/AndroidModule.scala

```diff
@@ -582,6 +571,56 @@ trait AndroidModule extends JavaModule { outer =>
    */
   def androidNamespace: String
 
+  /**
+   * The package name where the BuildInfo.java file will be generated.
+   * Defaults to [[androidNamespace]].
+   */
+  def androidBuildInfoPackageName: String = androidNamespace
+
+  /**
+   * The members to include in the generated BuildConfig.java file.
+   * Format is "type NAME = value"
+   */
+  def androidBuildConfigMembers: T[Seq[String]] = Task {
+    val buildType = if (androidIsDebug()) "debug" else "release"
+    Seq(
+      s"boolean DEBUG = ${androidIsDebug()}",
+      s"""String BUILD_TYPE = "$buildType"""",
+      s"int VERSION_CODE = ${androidVersionCode()}",
+      s"""String VERSION_NAME = "${androidVersionName()}""""
+    )
+  }
+
+  /**
+   * Generates a BuildConfig.java file in the [[androidBuildInfoPackageName]] package
+   * This is a basic implementation of AGP's build config feature!
+   */
+  def generatedBuildConfig: T[PathRef] = Task {
+    val parsedMembers: Seq[String] = androidBuildConfigMembers().map { member =>
+      s"public static final $member;"
+    }
+    val content: String =
+      s"""
+         |package $androidBuildInfoPackageName;
+         |public final class BuildConfig {
+         |  ${parsedMembers.mkString("\n  ")}
+         |}
+          """.stripMargin
+
+    val destination = Task.dest / "source" / os.SubPath(androidBuildInfoPackageName.replace(
+      ".",
+      "/"
+    )) / "BuildConfig.java"
+
+    os.write(destination, content, createFolders = true)
+
+    PathRef(destination)
+  }
+
+  override def generatedSources: T[Seq[PathRef]] = Task {
+    super.generatedSources() ++ Seq(generatedBuildConfig())
```

Comment:
this [is configurable in AGP right](https://github.com/vaslabs-ltd/compose-samples-with-mill/blob/main/Jetcaster/mobile/build.gradle.kts#L87)? I mean creating a build config